### PR TITLE
Google button

### PR DESCRIPTION
**Pull request by byeongjae-kang**
## Description

Please describe and/or use bullet points to to explain what this PR is about.
### Create our own google button 
 - setup fontFamily
 - update Typography
 - update Button
 - create GoogleIcon
 - create GoogleButton
 - add testing and story

## Jira Ticket

Please copy and paste Jira issue link here.
https://fridge-manager.atlassian.net/browse/IRM-188

## Images or Videos for review

Please try to add images or videos that are helpful to understand what this PR is about.

https://github.com/byeongjae-kang/ate-fe/assets/82554049/d28d84f7-4363-4b7e-881e-708137cc55a0

## Check List

Please add `x` in the list below where applicable.

- [x] Test added or updated?
- [ ] Documentation added or updated? If so, please share the link below
- [x] Important decisions made with team?

### CODE REVIEW

File: src/components/GoogleButton/GoogleButton.stories.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      >
+        <Story />
+      </View>
+    )
+  ]
+};
+
+export default Meta;
+
+export const Basic = {};
```
Comment (hncpyj) [inline src/components/GoogleButton/GoogleButton.stories.js:30]: What does this do?

File: src/components/GoogleButton/GoogleButton.styles.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export const styles = StyleSheet.create({
+  button: {
+    gap: 8,
+    padding: 12,
+    flexDirection: "row",
+    justifyContent: "center",
+    alignItems: "center",
+    borderWidth: 1,
+    // TODO: we do not have this in our DS
```
Comment (hncpyj) [inline src/components/GoogleButton/GoogleButton.styles.ts:12]: oh😂 this should have been on ds

File: src/components/GoogleButton/GoogleButton.styles.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    borderColor: "#D9D9D9",
+    borderRadius: 30,
+    backgroundColor: COLORS.white,
+    width: "100%"
+  },
+  pressed: {
+    // TODO: we do not have this in our DS
+    backgroundColor: "#EEEEEE"
```
Comment (hncpyj) [inline src/components/GoogleButton/GoogleButton.styles.ts:20]: 👍

File: src/context/auth-context/google-sign-in.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ * the package does not support Expo Go, so I had mock it
+ * without this Expo Go cannot be run at all
+ *
+ * TODO: when we do not need expo-go please remove this tile
+ */
+
+interface User {
+  user: {
+    id: string;
+    name: string | null;
```
Comment (hncpyj) [inline src/context/auth-context/google-sign-in.ts:14]: The full name is also used. I need to add the full name to our database as well.

File: src/context/auth-context/google-sign-in.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    email: string;
+    photo: string | null;
```
Comment (hncpyj) [inline src/context/auth-context/google-sign-in.ts:16]: Is this data like url?

File: src/context/auth-context/google-sign-in.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    };
+  }
+  async signOut() {}
+
+  async hasPlayServices() {}
+}
+
+let GoogleSigninSingleton = new FakeGoogleSignin();
+
+if (Constants.appOwnership !== "expo") {
```
Comment (hncpyj) [inline src/context/auth-context/google-sign-in.ts:53]: nice!

**Review by hncpyj — APPROVED** [review]

File: src/components/GoogleButton/GoogleButton.stories.js (hunk 1 above)
Comment (byeongjae-kang) [reply src/components/GoogleButton/GoogleButton.stories.js:30]: The basic is one of Meta data such as props and Storybook creates component with it behind the scene :)

File: src/components/GoogleButton/GoogleButton.styles.ts (hunk 2 above)
Comment (byeongjae-kang) [reply src/components/GoogleButton/GoogleButton.styles.ts:12]: I know right?

File: src/context/auth-context/google-sign-in.ts (hunk 4 above)
Comment (byeongjae-kang) [reply src/context/auth-context/google-sign-in.ts:14]: Will keep that it mind!!

File: src/context/auth-context/google-sign-in.ts (hunk 5 above)
Comment (byeongjae-kang) [reply src/context/auth-context/google-sign-in.ts:16]: yeah

**Pull request merged**